test/two_links_moving_joint-launch.py

```diff
@@ -0,0 +1,56 @@
+import os
+import unittest
+
+from launch import LaunchDescription
+import launch
+from launch_ros.actions import Node
+import launch_testing
+from launch.substitutions import LaunchConfiguration
+from launch_ros.substitutions import FindPackageShare
+
+
+def generate_test_description(ready_fn):
+    test_exe_arg = launch.actions.DeclareLaunchArgument(
+        'test_exe',
+        description='Path to executable test',
+    )
+
+    process_under_test = launch.actions.ExecuteProcess(
+        cmd=[launch.substitutions.LaunchConfiguration('test_exe')],
+        output='screen',
+    )
+
+    urdf_file = os.path.join(os.path.dirname(__file__), 'two_links_moving_joint.urdf')
```

Comment:
Correct me if I'm wrong, but it looks like this is the only difference with the other launch_testing file. Can we consolidate and make the urdf file name a launch argument?

Reply:
It's not a bad idea.  I'll follow-up a little later on this.

Reply:
So I tried to do this, but I don't think it is possible (or at least, I don't know how to do it with the launch_testing API).  The problem is that inside the launch file, I need to do something like:

```
urdf_path = os.path.join(os.path.dirname(__file__), LaunchConfiguration('urdf_file').expand())
with open(urdf_path) as infp:
    robot_desc = infp.read()
params = {'robot_description': robot_desc}
```

Unfortunately, the `expand()` method on `LaunchConfiguration` just doesn't exist.  There is `perform()`, but that needs some sort of context that I don't know how to get.  Without that, I can't read in the robot_description that I need.  So I'm going to give up on this for now.

Any ideas on how to accomplish this are welcome!

Reply:
I see the problem. Here's a [similar question](https://answers.ros.org/question/340705/access-launch-argument-in-launchfile-ros2/) with not quite the answer we're looking for. I feel like it should be possible; e.g. wrap the LaunchConfiguration substitution in another Substitution that does the file read. I don't have bandwidth atm to look into, so I'm happy to move forward with this PR as-is and if we figure it out we can revisit it.